app/controllers/architectures_controller.rb

```diff
@@ -39,4 +55,33 @@ def destroy
       process_error
     end
   end
+
+  private
+
+  def controller_permission
+    'architectures'
+  end
```

Comment:
Why is this needed? We already have a `controller_permission` in the `ApplicationController` which should derive it correctly. Is that broken?